### PR TITLE
fix(mobile): add safe-area-inset on app layout and specific views

### PR DESCRIPTION
**Pull request by bemble**
Also handle safe area in bottom bars

## Proposed change

Add safe-area-inset on layout and other views to fix content to close to mobile screen edge (like iPhones without touch button).

Before (iPhone X):

https://user-images.githubusercontent.com/397503/194511489-928a744e-6074-4a27-b90d-4de3779895b2.MP4

After (iPhone X):

https://user-images.githubusercontent.com/397503/194511657-04b72a80-ba4d-46c6-bbdb-9704902d5294.MP4

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/developer-tools/event/developer-tools-event.js [hunk 1]
```diff
@@ -25,7 +25,10 @@ class HaPanelDevEvent extends EventsMixin(LocalizeMixin(PolymerElement)) {
       <style include="ha-style iron-flex iron-positioning"></style>
       <style>
         .content {
-          padding: 16px;
```
Comment (bramkragten) [inline src/panels/developer-tools/event/developer-tools-event.js:28]: Not everything supports things like `env`, leave this in, and if it is supported the rule below will overrule it.

File: src/panels/developer-tools/service/developer-tools-service.ts [hunk 2]
```diff
@@ -391,12 +391,19 @@ class HaPanelDevService extends LitElement {
       haStyle,
       css`
         .content {
-          padding: 16px;
```
Comment (bramkragten) [inline src/panels/developer-tools/service/developer-tools-service.ts:394]: keep this

File: src/panels/developer-tools/service/developer-tools-service.ts [hunk 3]
```diff
@@ -391,12 +391,19 @@ class HaPanelDevService extends LitElement {
...
+          padding: calc(16px + env(safe-area-inset-top))
+            calc(16px + env(safe-area-inset-right))
+            calc(16px + env(safe-area-inset-bottom))
+            calc(16px + env(safe-area-inset-left));
           max-width: 1200px;
           margin: auto;
         }
         .button-row {
           padding: 8px 16px;
+          padding: calc(16px + env(safe-area-inset-top))
```
Comment (bramkragten) [inline src/panels/developer-tools/service/developer-tools-service.ts:403]: you changed the padding from 8 to 16px?

File: src/panels/developer-tools/statistics/developer-tools-statistics.ts [hunk 4]
```diff
@@ -326,7 +326,10 @@ class HaPanelDevStatistics extends SubscribeMixin(LitElement) {
       haStyle,
       css`
         .content {
-          padding: 16px;
```
Comment (bramkragten) [inline src/panels/developer-tools/statistics/developer-tools-statistics.ts:329]: keep

File: src/panels/developer-tools/template/developer-tools-template.ts [hunk 5]
```diff
@@ -248,7 +248,10 @@ class HaPanelDevTemplate extends LitElement {
         }
 
         .content {
-          padding: 16px;
```
Comment (bramkragten) [inline src/panels/developer-tools/template/developer-tools-template.ts:251]: keep

File: src/panels/developer-tools/yaml_configuration/developer-yaml-config.ts [hunk 6]
```diff
@@ -214,7 +214,10 @@ export class DeveloperYamlConfig extends LitElement {
         }
 
         .content {
-          padding: 28px 20px 16px;
```
Comment (bramkragten) [inline src/panels/developer-tools/yaml_configuration/developer-yaml-config.ts:217]: This one is still removed

Comment (bemble) [reply src/panels/developer-tools/yaml_configuration/developer-yaml-config.ts:217]: Oups, I forgot it. I used 16px as a base for padding, to keep the same spacing logic.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**